### DISCUSSION TITLE
关于waline 服务端版本号与升级问题

### DISCUSSION THREAD

**Discussion by appotry** [walinejs/waline#1440] [Q&A] [answered]
部署在vercel 上面，网页端找遍了没看到哪里有服务端版本号

是我没找对地方么？

目前这种vercel 服务端部署方式，部署成功后，服务端是不会升级的！ 手动触发重新部署可以升级版本

那么就有问题了，没办法知道服务端是否过期，是否有bug或者功能需要更新版本，需要一个版本升级提示

**Comment by elkan1788** [comment]
打开你的 waline ui 操作界面，然后按下 F12 键，可以在控制台上面看到服务端的版本号输出，效果参考如下：

![image](https://user-images.githubusercontent.com/1217096/193399866-e7a6c170-3c5f-4ceb-9708-4f7c1f93f570.png)

**Comment by lizheming** [comment]
https://blog.ccknbc.cc/posts/waline-comments-system-deployment-logs/ 可以参考下 CC 老师的方案

**Comment by appotry** [comment, marked as answer]
@lizheming ccknbc的方式明白原理了，利用action 修改任何一点文件即可以。算是被动触发重新部署方式

## 主动触发部署方式思路：

说一个个人设想的思路，自动部署一般都有有手动触发api， waline 客户端或者服务端启动的时候会去github api 查询最新版本，如果发现更新的版本就触发部署api。 

或者给管理员账户发个提醒邮件，或者再管理面板放一个更新提示，通过github api判断版本是否需要更新

这样是不是更加简单易用

**Reply by Mister-Hope** [reply]
这个建议我觉得可以，@waline/admin在启动的时候可以增加去 npm 拉取 @waline/vercel 最新版本并比对的过程，然后显示弹窗进行提示。

当然，用户也可以使用 Github Workflow 提供的 cron 表达式创建定时任务每日检查更新。 @lizheming 上面的想法觉得怎么样？

**Reply by Mister-Hope** [reply]
@lizheming 这个需求有 Feedback 么，检查可以发生在 admin 登陆时。

**Reply by lizheming** [reply]
@Mister-Hope 现在后台已经有更新提醒了
![image](https://user-images.githubusercontent.com/424491/203943519-824124a4-cf6e-41fc-b438-81f65b97ad11.png)
